### PR TITLE
[Bug] deploy job on k8s session mode get job state bug fixed

### PR DESCRIPTION
**Pull request by wolfboys**
[Bug] deploy job on k8s session mode get job state bug fixed

### CODE REVIEW

File: streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/DeployResponse.scala [hunk 1]
```diff
@@ -17,4 +17,4 @@
 
 package org.apache.streampark.flink.client.bean
 
-case class DeployResponse(address: String, clusterId: String)
+case class DeployResponse(address: String = null, clusterId: String = null, error: Throwable = null)
```
Comment (caicancai) [inline streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/DeployResponse.scala:20]: I need to make sure if there are any changes related to v2 here

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala [hunk 2]
```diff
@@ -204,18 +197,17 @@ object KubernetesNativeSessionClient extends KubernetesNativeClientTrait with Lo
...
-    } finally {
-      Utils.close(client)
+    Try(
+      clusterDescriptor
+        .retrieve(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID))
+        .getClusterClient
+    ) match {
+      case Failure(e) => ShutDownResponse(e)
+      case Success(c) =>
+        Try(c.shutDownCluster()) match {
```
Comment (caicancai) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala:207]: good

**Comment by caicancai** [comment]
cc @Al-assad @MonsterChenzhuo If you have time, could you please review this PR? thank you

File: streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/DeployResponse.scala (hunk 1 above)
Comment (wolfboys) [reply streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/DeployResponse.scala:20]: > I need to make sure if there are any changes related to v2 here

Only solve the issues in the current branch, it won't affect v2.

Comment (caicancai) [reply streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/DeployResponse.scala:20]: thanks

**Review by caicancai — APPROVED** [review]

**Comment by caicancai** [comment]
LGTM

**Comment by caicancai** [comment]
Maybe we should create an issue for this PR when merging it into the dev branch in the future.

**Comment by wolfboys** [comment]
> Maybe we should create an issue for this PR when merging it into the dev branch in the future.

Yes, these improvements will be submitted to the dev branch

**Pull request merged**